### PR TITLE
LPS-142279 test setting shielded container lib dir in InitUtil

### PR DESCRIPTION
**Pull request by julschong**

### CODE REVIEW

**Comment by julschong** [comment]
ci:test:sf

**Comment by julschong** [comment]
ci:test:relevant

**Comment by julschong** [comment]
ci:test

**Comment by julschong** [comment]
ci:test:environment

**Comment by julschong** [comment]
ci:test:environments

**Comment by julschong** [comment]
@catarineaguiar Hi Catarine, Could you please help me scan this test result?
Thank you!

**Comment by vicnate5** [comment]
@kyle-miho can you check this one?

**Comment by kyle-miho** [comment]
reading results rn

**Comment by julschong** [comment]
ci:test:sf

**Comment by julschong** [comment]
ci:test

**Comment by julschong** [comment]
ci:test:environments

**Comment by julschong** [comment]
@catarineaguiar Hi Catarine, Could you please help me scan this test result?
Thank you!

File: portal-impl/src/com/liferay/portal/util/PropsUtil.java [hunk 1]
```diff
@@ -383,29 +393,27 @@ private static String _getLibDir(Class<?> clazz) {
...
-				0, portalWebDir.length() - 31);
-		}
+			if (_log.isDebugEnabled()) {
+				_log.debug("Portal web directory " + portalWebDir);
+			}
 
-		if (_log.isDebugEnabled()) {
-			_log.debug("Portal web directory " + portalWebDir);
+			SystemProperties.set(
+				PropsKeys.LIFERAY_WEB_PORTAL_DIR, portalWebDir);
```
Comment (julschong) [inline portal-impl/src/com/liferay/portal/util/PropsUtil.java:414]: only set LIFERAY_WEB_PORTAL_DIR when portalShieldedContainerLibDirProperty is set

File: portal-impl/src/com/liferay/portal/util/PropsUtil.java [hunk 2]
```diff
@@ -367,8 +379,6 @@ private static String _getLibDir(Class<?> clazz) {
 
 		// Portal shielded container lib directory
 
-		String portalShieldedContainerLibDir = _getLibDir(PropsUtil.class);
-
```
Comment (julschong) [inline portal-impl/src/com/liferay/portal/util/PropsUtil.java:371]: Removing _getLibDir for portalShieldedContainerLibDir, this was using in DBBuilder.
We moved this part of logic into InitUtil.
If this is not set, ant all throws compile error

File: portal-impl/src/com/liferay/portal/util/InitUtil.java [hunk 3]
```diff
@@ -148,6 +152,33 @@ public static synchronized void init() {
...
+		}
+
+		int pos = path.lastIndexOf("!/");
+
+		pos = path.lastIndexOf(CharPool.SLASH, pos);
+
+		SystemProperties.set(
+			PropsKeys.LIFERAY_SHIELDED_CONTAINER_LIB_PORTAL_DIR,
+			path.substring(0, pos + 1));
+
```
Comment (julschong) [inline portal-impl/src/com/liferay/portal/util/InitUtil.java:181]: Setting LIFERAY_SHIELDED_CONTAINER_LIB_PORTAL_DIR here

File: portal-impl/src/com/liferay/portal/util/PropsUtil.java [hunk 4]
```diff
@@ -356,7 +368,7 @@ private static String _getLibDir(Class<?> clazz) {
 
 		// Global shared lib directory
 
-		String globalSharedLibDir = _getLibDir(Servlet.class);
+		String globalSharedLibDir = _getLibDir(Servlet.class); // TODO: Get rid of _getLibDir
```
Comment (julschong) [inline portal-impl/src/com/liferay/portal/util/PropsUtil.java:371]: Still need a way to remove this

**Pull request closed without merging**